platform/darwin/src/http_file_source.mm

```diff
@@ -215,14 +215,19 @@ void cancel() {
         }
 
         [req addValue:impl->userAgent forHTTPHeaderField:@"User-Agent"];
-
+        
+        if (resource.kind == mbgl::Resource::Kind::Tile) {
+            [[MGLNetworkConfiguration sharedManager] startDownloadEvent:url.relativePath type:@"tile"];
+        }
+        
         request->task = [impl->session
             dataTaskWithRequest:req
               completionHandler:^(NSData* data, NSURLResponse* res, NSError* error) {
                 if (error && [error code] == NSURLErrorCancelled) {
+                    [[MGLNetworkConfiguration sharedManager] cancelDownloadEvent:res.URL.relativePath];
                     return;
                 }
-
+                [[MGLNetworkConfiguration sharedManager] stopDownloadEvent:res.URL.relativePath];
```

Comment:
We're not currently distinguishing between success and error here. What does Android do here?